### PR TITLE
add docker-socket-proxy as option

### PR DESCRIPTION
**Pull request by szaimen**
Close https://github.com/nextcloud/all-in-one/discussions/3230

- [ ] needs https://github.com/nextcloud-releases/all-in-one/pull/43

todo:
- [x] add app-ecosystem app into nextcloud if the container is enabled

### CODE REVIEW

File: Containers/docker-socket-proxy/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    POST=0 \
+    SECRETS=0 \
+    SERVICES=1 \
+    SESSION=0 \
+    SOCKET_PATH=/var/run/docker.sock \
+    SWARM=0 \
+    SYSTEM=0 \
+    TASKS=0 \
+    VERSION=1 \
+    VOLUMES=1
```
Comment (szaimen) [inline Containers/docker-socket-proxy/Dockerfile:36]: @andrey18106 @bigcat88 would be good to know which endpoints you need :)

File: php/containers.json [hunk 2]
```diff
@@ -639,6 +641,29 @@
...
+      "image": "nextcloud/aio-docker-socket-proxy",
+      "init": true,
+      "internal_port": "2375",
+      "environment": [
+        "TZ=%TIMEZONE%"
+      ],
+      "volumes": [
+        {
+          "source": "%WATCHTOWER_DOCKER_SOCKET_PATH%",
+          "destination": "/var/run/docker.sock",
```
Comment (Zoey2936) [inline php/containers.json:657]: https://github.com/nextcloud/all-in-one#are-there-known-problems-when-selinux-is-enabled

File: Containers/nextcloud/entrypoint.sh [hunk 3]
```diff
@@ -732,5 +732,20 @@ else
...
 fi
 
+# Docker socket proxy
+if [ "$DOCKER_SOCKET_PROXY_ENABLED" = 'yes' ]; then
+    if ! [ -d "/var/www/html/custom_apps/app_ecosystem_v2" ]; then
+        php /var/www/html/occ app:install app_ecosystem_v2
+    elif [ "$(php /var/www/html/occ config:app:get app_ecosystem_v2 enabled)" != "yes" ]; then
+        php /var/www/html/occ app:enable app_ecosystem_v2
+    elif [ "$SKIP_UPDATE" != 1 ]; then
+        php /var/www/html/occ app:update files_antivirus
```
[the remark this answers is not in the document]
Comment (szaimen) [reply Containers/nextcloud/entrypoint.sh]: jep, fixed :)

File: php/containers.json (hunk 2 above)
Comment (szaimen) [reply php/containers.json:657]: Right, I forgot this. But I guess it would probably also produce problems with watchtower?

Comment (szaimen) [reply php/containers.json:657]: I suppose we should probably set the flag for these containers then by default...

Comment (Zoey2936) [reply php/containers.json:657]: maybe yes

Comment (szaimen) [reply php/containers.json:657]: https://github.com/nextcloud/all-in-one/issues/3239

File: Containers/docker-socket-proxy/Dockerfile (hunk 1 above)
Comment (andrey18106) [reply Containers/docker-socket-proxy/Dockerfile:36]: Services, Events - not needed. The rest is fine

**Comment by szaimen** [comment]
Done @andrey18106 :)

File: php/templates/containers.twig [hunk 4]
```diff
@@ -574,6 +574,11 @@
                         {% else %}
                             {#<input type="checkbox" id="onlyoffice" name="onlyoffice"><label for="onlyoffice">OnlyOffice</label><br>#}
                         {% endif %}
+                        {% if is_docker_socket_proxy_enabled == true %}
+                            <input type="checkbox" id="docker-socket-proxy" name="docker-socket-proxy" checked="checked"><label for="docker-socket-proxy">Docker Socket Proxy (needed for app-ecosystem v2)</label><br><br>
+                        {% else %}
+                            <input type="checkbox" id="docker-socket-proxy" name="docker-socket-proxy"><label for="docker-socket-proxy">Docker Socket Proxy (needed for app-ecosystem v2)</label><br><br>
```
[the remark this answers is not in the document]
Comment (szaimen) [reply php/templates/containers.twig]: done

File: Containers/docker-socket-proxy/Dockerfile (hunk 1 above)
Comment (szaimen) [reply Containers/docker-socket-proxy/Dockerfile]: done

**Review by andrey18106 — APPROVED** [review]

**Comment by andrey18106** [comment]
> This is now released with v7.1.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

Tested with the latest develop channel - looks good.

**Pull request merged**